pkg/auth/dependency/interaction/flows/password.go

```diff
@@ -0,0 +1,79 @@
+package flows
+
+import (
+	"github.com/skygeario/skygear-server/pkg/auth/dependency/authenticator"
+	"github.com/skygeario/skygear-server/pkg/auth/dependency/interaction"
+	"github.com/skygeario/skygear-server/pkg/core/authn"
+)
+
+type PasswordFlow struct {
+	Interactions InteractionProvider
+}
+
+func (f *PasswordFlow) ChangePassword(userID string, OldPassword string, newPassword string) error {
+	i, err := f.Interactions.NewInteractionUpdateAuthenticator(&interaction.IntentUpdateAuthenticator{
+		Authenticator: authenticator.Spec{
+			Type: authn.AuthenticatorTypePassword,
+		},
+	}, "", userID, map[string]string{
+		authenticator.AuthenticatorStatePasswordCheckOldPassword: OldPassword,
```

Comment:
We should avoid storing sensitive data (i.e. raw password) in the interaction state, since it would be persisted to database.

Instead, we may do it like this:
1. Create interaction using `IntentUpdateAuthenticator`, providing `Authenticator` & `Secret` (old)
2. The authenticator is resolved and secret is verified before returning an `Interaction`. Secret is not persisted in interaction state.
3. User enter new secret.
4. Perform `ActionSetupAuthenticator` with the new secret.

Reply:
Thanks for catching this! Updated in 417fda4